src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnectionOptions.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Security;
+
+namespace System.Net.Quic;
+
+/// <summary>
+/// Shared options for both client (outbound) and server (inbound) <see cref="QuicConnection" />.
+/// </summary>
+public abstract class QuicConnectionOptions
+{
+    /// <summary>
+    /// Prevent sub-classing.
+    /// </summary>
+    internal QuicConnectionOptions()
+    { }
+
+    /// <summary>
+    /// Limit on the number of bidirectional streams the remote peer connection can create on an open connection.
+    /// Default 0 for client and 100 for server connection.
+    /// </summary>
+    public int MaxBidirectionalStreams { get; set; }
+
+    /// <summary>
+    /// Limit on the number of unidirectional streams the remote peer connection can create on an open connection.
+    /// Default 0 for client and 10 for server connection.
+    /// </summary>
+    public int MaxUnidirectionalStreams { get; set; }
+
+    /// <summary>
+    /// Idle timeout for connections, after which the connection will be closed.
+    /// Default <see cref="TimeSpan.Zero"/> means underlying implementation default idle timeout.
+    /// </summary>
+    public TimeSpan IdleTimeout { get; set; } = TimeSpan.Zero;
```

Comment:
Zero seems a bit inconsistent for this.  With SocketsHttpHandler, if you set its idle timeout to 0, that means connections won't be pooled since they'll be torn down the moment they're added to the pool.

Reply:
We should discuss this API review next Tuesday. I summarized the reasoning here: https://github.com/dotnet/runtime/issues/68902#issuecomment-1170128344, but I'm open to changes. The zero meaning here is trying to be consistent with `ListenBacklog = 0` meaning use whatever is the best for the current platform.
Apart from this, 0 in msquic means infinite (for which we already have -1), and we don't have a mechanism to automatically close the connection after ... I don't even know what would be single use in this space, one outbound stream?

Moreover, there's no explicit pooling, it's a timeout to automatically tear down a connection is no data flow on it. So it's slightly different concept than SocketsHttpHandler.